pkg/operator/controller/controller_dns.go

```diff
@@ -35,11 +35,6 @@ func (r *reconciler) ensureDNS(ci *ingressv1alpha1.ClusterIngress, service *core
 func desiredDNSRecords(ci *ingressv1alpha1.ClusterIngress, service *corev1.Service, dnsConfig *configv1.DNS) ([]*dns.Record, error) {
 	records := []*dns.Record{}
 
-	// If no service exists, no DNS records should exist.
```

Comment:
Rationale for removing this? `ensureLoadBalancerService` still returns `nil` if no LB service should be created, which means `service` here can be `nil`...